### ISSUE TITLE
Fix S2325 FP: Incorrectly suggests making C# 14 extension-block members static (doesn’t compile)

### ISSUE THREAD

**Issue by HaikAsatryan** [SonarSource/sonar-dotnet#9782]
### Description

Rule **S2325** ("Methods and properties that don't access instance data should be static") is raised on **C# 14 / .NET 10 extension blocks** (new `extension(...) { ... }` syntax).

### Reproducer

For extension-block members, Sonar suggests:  
> Make 'GetAllowedCorsOrigins' a static method

But these members **cannot** be made `static` in an *instance receiver* extension block (`extension(IConfiguration configuration)`), so the suggested fix is invalid and (in my case) results in a compiler error.

Additionally, the flagged members **do** access instance data: they use the receiver variable (`configuration`) from the extension block.
```cs
internal static class ConfigurationExtensions
{
    private const string CorsOriginsConfigurationPath = "Security:AllowedCorsOrigins";
    private const string PersistentConfigurationPath = "PersistentStorage";

    extension(IConfiguration configuration)
    {
        internal string GetAllowedCorsOrigins()
        {
            var corsOrigins = configuration[CorsOriginsConfigurationPath];
            return corsOrigins ?? throw new InvalidOperationException("Allowed CORS origins are not configured.");
        }

        internal string GetRepositoryName()
        {
            var repositoryName = configuration[RepositoryNameConfigurationPath];
            return repositoryName ?? throw new InvalidOperationException("Repository name is not configured.");
        }
    }
}
```

### Product and Version

SonarAnalyzer.CSharp 10.15.0.120848

**Comment by alex-meseldzija-sonarsource** [comment]
Thanks for the report. 
We'll close this as a duplicate of https://github.com/SonarSource/sonar-dotnet/issues/9772 and [this from our community forum](https://community.sonarsource.com/t/fp-s2325-does-report-on-c-14-extensions/151936).

This is fixed in [this commit](https://github.com/SonarSource/sonar-dotnet/commit/52e1158fd460ee3a227819771c33ca71819dca50) and will be included in the next release.
